bp_common/syn/tcl/vivado_synth.tcl

```diff
@@ -23,18 +24,28 @@ foreach x $f {
       set temp [string trimleft $x $trimchars]
       set expanded [subst $temp]
       lappend dir_list $expanded
+    } elseif {[string match "*bsg_mem_1rw_sync_mask_write_bit.v" $x]} {
+      set replace_hard "$BASEJUMP_STL_DIR/hard/ultrascale_plus/bsg_mem/bsg_mem_1rw_sync_mask_write_bit.v"
+      set expanded [subst $replace_hard]
+      lappend flist $expanded
+      puts $expanded
     } else {
       set expanded [subst $x]
       lappend flist $expanded
     }
   }
 }
 
+puts $flist
+
 set_part $PART
 read_verilog -sv $flist
 read_xdc design.xdc
 
-synth_design -top wrapper -part $PART -include_dirs $dir_list
+synth_design -top wrapper -part $PART -include_dirs $dir_list -flatten_hierarchy none
```

Comment:
Does this fully preserve hierarchy? i.e. are module boundaries preseved?

Reply:
Yes, all the modules have the proper interface signals. Would be helpful for debugging

Reply:
Interesting, are ports removed if they're optimized out (that how DC works)?  Makes me think there'd be a way to maintain port-only binds... But don't worry for this PR, parameterizing out the binds is fine (although it's very unfortunate that we lose cosim this way)

Reply:
According to the documentation, setting flatten_hierarchy to none keeps the exact same hierarchy. I saw a few modules while trying to debug and they had the correct ports. I'm not sure if ports are optimized out, the doc only mentions that the hierarchy is maintained

Reply:
Cool!